### DISCUSSION TITLE
Dificulty with post_report

### DISCUSSION THREAD

**Discussion by vgarciasouza** [denisneuf/python-amazon-ad-api#142] [Q&A]
Hey, been trying to use the post_report method and getting the error:

`  File "...\sync_ads_data.py", line 130, in <module>
    sync_local_ads_files()
  File "...\sync_ads_data.py", line 117, in sync_local_ads_files
    report_file = download_report_to_csv(
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "...\ads_reports.py", line 67, in download_report_to_csv
    result = client.post_report(body=req_body, debug=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\ad_api\base\helpers.py", line 21, in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\ad_api\api\reports.py", line 38, in post_report
    return self._request(kwargs.pop('path'), data=kwargs.pop('body'), params=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\ad_api\base\client.py", line 278, in _request
    return self._check_response(res)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\ad_api\base\client.py", line 300, in _check_response
    data = json.loads(str_content)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Python311\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Python\Python311\Lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "...\Python311\Lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

Here's my code:

```
name: str,
start_date: str,
end_date: str,
group_by: str,
columns: list,
report_type: list,

req_body = {
        "name": name,
        "startDate": start_date,
        "endDate": end_date,
        "configuration": {
            "adProduct": "SPONSORED_PRODUCTS",
            "groupBy": group_by,
            "columns": columns,
            "reportTypeId": report_type,
            "timeUnit": "DAILY",
            "format": "GZIP_JSON"
        }
    }

    client = create_ads_client()

    # Generate the report from ads API
    print(f"\nGenerating report {name}...")
    result = client.post_report(body=req_body)
```

Any ideas on why it is happenning? Thanks!

**Comment by denisneuf** [comment]
Try this request:

group_by: str, Is a list not str  ["advertiser"],
report_type: list, is a str not a list "spAdvertisedProduct"

```
    request = \
        {
            "startDate": "2022-11-01",
            "endDate": "2022-11-01",
            "configuration": {
                "adProduct": "SPONSORED_PRODUCTS",
                "groupBy": ["advertiser"],
                "columns": ["impressions", "clicks", "cost", "campaignStatus", "advertisedAsin", "date"],
                "reportTypeId": "spAdvertisedProduct",
                "timeUnit": "DAILY",
                "format": "GZIP_JSON"
            }
        }
```

Let me know if it works for you. If not, post the full code with imports, activate Debug and let me know the api version you are using.

**Comment by vgarciasouza** [comment]
Actually I missed the .env file to set AWS_ENV = PRODUCTION.

Also I needed to pass the body inside a json.dumps()

 result = client.post_report(body=json.dumps(req_body))

With that i was able to post the reports and get the responses.

Thank you!
